src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react';
+import { TPAComponentsConsumer } from '../TPAComponentsConfig';
+
+import styles from './Modal.st.css';
+import { IconButton } from '../IconButton';
+import { Close } from '../../assets/icons';
+import { DATA_HOOKS } from './constants';
+
+export interface ModalProps {
+  /**
+   * Describing if the modal should be shown or not.
+   */
+  isOpen: boolean;
+
+  /**
+   * Function that will be run when the modal is requested to be closed (by clicking on overlay)
+   * Note: It is not called if isOpen is changed by other means.
+   */
+  onRequestClose(): void;
+
+  /**
+   * Describing if the modal should be shown in full screen or not.
+   */
+  inFullScreen?: boolean;
+
+  /**
+   * Describing if the modal should be shown with close button or not.
+   */
+  withCloseButton?: boolean;
+
+  /**
+   * Describing if the modal should be shown with dark background overlay or not.
+   */
+  withBackground?: boolean;
+  aria?: {
+    labeledBy?: string;
+  };
+}
+
+interface DefaultProps {
+  inFullScreen: boolean;
+  withCloseButton: boolean;
+  withBackground: boolean;
+  aria: {
+    labeledBy: string;
+  };
+}
+
+interface State {}
+
+/** The modal component provides a solid foundation for creating dialogs, popovers, lightboxes, or whatever else. */
+export class Modal extends React.Component<ModalProps, State> {
+  static displayName = 'Modal';
+  static defaultProps: DefaultProps = {
+    inFullScreen: false,
+    withCloseButton: true,
+    withBackground: true,
+    aria: {
+      labeledBy: 'dialog-label',
+    },
```

Comment:
In addition, we shouldn't have a default value for this, since this is something we really want our consumers to use.
And since this also needs to be translated to fit the users language, we should let consumers configure this.